### PR TITLE
zephyr: Improve signature kconfig

### PR DESCRIPTION
**Pull request by hakonfam**

### CODE REVIEW

**Comment by nvlsianpu** [comment]
cc @MarekPieta

**Review by nvlsianpu — APPROVED** [review]

**Review by utzig — APPROVED** [review]

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -68,6 +68,7 @@ config SINGLE_APPLICATION_SLOT
 
 choice
 	prompt "Signature type"
+	default BOOT_SIGNATURE_TYPE_ECDSA_P256 if HAS_HW_NRF_CC310
```
Comment (MarekPieta) [inline boot/zephyr/Kconfig:71]: We could select using `CC310` by default for `BOOT_SIGNATURE_TYPE_ECDSA_P256` if available. Currently tinycrypt is used by default (even if the board supports `CC310`).

Comment (hakonfam) [reply boot/zephyr/Kconfig:71]: I'm not sure I understand what you mean. Do you mean that CC310 should be the backend for all types of signature types when the HW is available?

Comment (utzig) [reply boot/zephyr/Kconfig:71]: > I'm not sure I understand what you mean. Do you mean that CC310 should be the backend for all types of signature types when the HW is available?

He mentioned `BOOT_SIGNATURE_TYPE_ECDSA_P256`, not all types.

Comment (MarekPieta) [reply boot/zephyr/Kconfig:71]: Sorry if something was not clear:
If CC310 HW is available, we can use CC310 as a default backend for `BOOT_SIGNATURE_TYPE_ECDSA_P256 `:

```
if BOOT_SIGNATURE_TYPE_ECDSA_P256
choice
	prompt "Ecdsa implementation"
	default BOOT_CC310 if HAS_HW_NRF_CC310
	default BOOT_TINYCRYPT
```
This is already added to nrf fork of MCUBoot bootloader repository.

Comment (hakonfam) [reply boot/zephyr/Kconfig:71]: As far as I can see this is what we will get now in both upstream and downstream. This due to what is shown here:

```
config BOOT_CC310
	bool "Use CC310"		bool "Use CC310"
	select BOOT_USE_NRF_CC310_BL if HAS_HW_NRF_CC310		select BOOT_USE_NRF_CC310_BL if HAS_HW_NRF_CC310
	select NRF_CC310_BL if HAS_HW_NRF_CC310		select NRF_CC310_BL if HAS_HW_NRF_CC310
	select NRFXLIB_CRYPTO if SOC_FAMILY_NRF		select NRFXLIB_CRYPTO if SOC_FAMILY_NRF
	select BOOT_USE_CC310		select BOOT_USE_CC310
```

Comment (hakonfam) [reply boot/zephyr/Kconfig:71]: I see your point, check it now @MarekPieta !

Comment (MarekPieta) [reply boot/zephyr/Kconfig:71]: Thank you. Approved.

**Review by MarekPieta — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@looks like this patch reveled problem of missing nrfxlib while cc310 in use.
https://github.com/JuulLabs-OSS/mcuboot/blob/master/boot/zephyr/CMakeLists.txt#L64-L69
this cause cmake build stage error:
``` Console
/home/anpu/codedev/zephyrproject/zephyr/cmake/extensions.cmake:1579 (message):
  No such file or directory: NRFXLIB_DIR:
  '/home/anpu/codedev/zephyrproject/zephyr/../nrfxlib'
Call Stack (most recent call first):
  CMakeLists.txt:66 (assert_exists)
```

<mcuboot>/ext/nrf/README.md contains description, but this is buried

**Comment by hakonfam** [comment]
> @hakonfam @MarekPieta
> 
> Looks like this patch reveled problem of missing nrfxlib while cc310 in use.
> https://github.com/JuulLabs-OSS/mcuboot/blob/master/boot/zephyr/CMakeLists.txt#L64-L69
> this cause cmake build stage error:
> 
> ```
> /home/anpu/codedev/zephyrproject/zephyr/cmake/extensions.cmake:1579 (message):
>   No such file or directory: NRFXLIB_DIR:
>   '/home/anpu/codedev/zephyrproject/zephyr/../nrfxlib'
> Call Stack (most recent call first):
>   CMakeLists.txt:66 (assert_exists)
> ```
> 
> /ext/nrf/README.md contains description, but this is buried
> 
> this made user-experience poor

Could we add a dependency to a kconfig option added by nrfxlib to solve this? Since a non-defined kconfig is considered not set.

**Pull request merged**